rosbag2_tests/test/rosbag2_tests/test_rosbag2_info_end_to_end.py

```diff
@@ -0,0 +1,93 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import unittest
+
+import launch
+import launch.actions
+import launch_testing
+import launch_testing.actions
+import launch_testing.asserts
+import launch_testing.markers
+
+import pytest
+
+
+# This is necessary to get unbuffered output from the process under test
+proc_env = os.environ.copy()
+proc_env['PYTHONUNBUFFERED'] = '1'
+
+rosbag_info_process = launch.actions.ExecuteProcess(
+    cmd=['ros2', 'bag', 'info', launch.substitutions.LaunchConfiguration('bag_file_path')],
+    output='screen', env=proc_env,
+)
+
+rosbag_info_process_wrong = launch.actions.ExecuteProcess(
+    cmd=['ros2', 'bag', 'info', '/path/to/non/existing/bag_file'],
+    output='screen', env=proc_env,
+)
+
+
+@pytest.mark.launch_test
+@launch_testing.markers.keep_alive
+def generate_test_description():
+    return launch.LaunchDescription([
+        launch.actions.DeclareLaunchArgument(
+            'bag_file_path',
+            description='Absolute path to a bag file.',
+            default_value='../resources/cdr_test'
+        ),
+        launch_testing.actions.ReadyToTest(),
+    ])
+
+
+class TestInfo(unittest.TestCase):
+
+    def test_ros2_bag_info(self, launch_service, proc_info, proc_output):
+        """Test terminating_proc without command line arguments."""
+        print('Running ros2 bag info')
+        with launch_testing.tools.launch_process(
+              launch_service, rosbag_info_process, proc_info, proc_output) as command:
+            assert command.wait_for_shutdown(timeout=3)
+            assert command.exit_code == launch_testing.asserts.EXIT_OK
+        # Using the SequentialStdout context manager asserts that the following stdout
+        # happened in the same order that it's checked
+        with launch_testing.asserts.assertSequentialStdout(proc_output, rosbag_info_process) as cm:
+            cm.assertInStdout('Files:             cdr_test.db3')
+            # cm.assertInStdout('Bag size:          .*B')
+            cm.assertInStdout('Storage id:        sqlite3')
+            cm.assertInStdout('Duration:          0.155s')
+            # cm.assertInStdout('Start:             Sep 18 2018 .*:.*:44.241 (1537282604.241)')
+            # cm.assertInStdout('End                Sep 18 2018 .*:.*:44.397 (1537282604.397)')
+            cm.assertInStdout('Messages:          7')
+            cm.assertInStdout(
+                ('Topic information: Topic: /test_topic | Type: test_msgs/BasicTypes '
+                 '| Count: 3 | Serialization Format: cdr'))
+            cm.assertInStdout(
+                ('Topic: /array_topic | Type: test_msgs/Arrays '
+                 '| Count: 4 | Serialization Format: cdr'))
+        # TODO(Karsten1987): Regex doesn't seem to work.
+        # launch_testing.asserts.assertInStdout(
+        #     proc_output, 'Files:\s+cdr_test.db3', rosbag_info_process)
```

Comment:
in connection to the above comments, https://github.com/ros2/launch/blob/master/launch_testing/launch_testing/asserts/assert_output.py#L46 mentions support for regex, but I wasn't able to find out a syntax for it.